test/offchainAssetVault/OffChainAssetVault.test.ts

```diff
@@ -150,4 +170,581 @@ describe("OffChainAssetVault", async function () {
+
+    //grant withdrawer role to alice
+    await vault.grantRole(await vault.WITHDRAWER(), alice.address);
+
+    const expectedShares = fixedPointMul(assets, id).add(1);
+
+    const shares = await vault["previewWithdraw(uint256,uint256)"](assets, id);
+
+    assert(
+      shares.eq(expectedShares),
+      `Wrong shares: expected ${expectedShares} got ${shares} `
+    );
+  });
+  it("PreviewRedeem sets correct assets", async function () {
+    const [vault] = await deployOffChainAssetVault();
+    const shares = ethers.BigNumber.from(100);
+
+    const signers = await ethers.getSigners();
+    const alice = signers[0];
+
+    const [receiptVault, asset, priceOracle] =
+      await deployERC20PriceOracleVault();
+
+    const id = await priceOracle.price();
+
+    const hasRoleDepositor = await vault.hasRole(
+      await vault.WITHDRAWER(),
+      alice.address
+    );
+
+    //Alice does not have role of withdrawer, so it should throw an error unless role is granted
+    assert(
+      !hasRoleDepositor,
+      `AccessControl: account ${alice.address.toLowerCase()} is missing role WITHDRAWER`
+    );
+
+    //grant withdrawer role to alice
+    await vault.grantRole(await vault.WITHDRAWER(), alice.address);
+
+    const expectedAssets = fixedPointDiv(shares, id);
+    const assets = await vault["previewRedeem(uint256,uint256)"](shares, id);
+
+    assert(
+      assets.eq(expectedAssets),
+      `Wrong assets: expected ${expectedAssets} got ${assets} `
+    );
+  });
+  it("Redeposit - should be receipt holder", async function () {
+    const signers = await ethers.getSigners();
+    const alice = signers[0];
+    const [vault] = await deployOffChainAssetVault();
+    const [receiptVault, asset, priceOracle] =
+      await deployERC20PriceOracleVault();
+    const shareRatio = await priceOracle.price();
+
+    const aliceReceiptBalance = await vault["balanceOf(address,uint256)"](
+      alice.address,
+      shareRatio
+    );
+
+    assert(aliceReceiptBalance.eq(0), `NOT_RECEIPT_HOLDER`);
+  });
+  it("Checks role for snapshotter", async function () {
+    const signers = await ethers.getSigners();
+    const alice = signers[0];
+    const [vault] = await deployOffChainAssetVault();
+
+    await assertError(
+      async () => await vault.snapshot(),
+      `AccessControl: account ${alice.address.toLowerCase()} is missing role ${await vault.ERC20SNAPSHOTTER()}`,
+      "failed to snapshot"
+    );
+  });
+  it("Snapshot event is emitted", async function () {
+    const signers = await ethers.getSigners();
+    const alice = signers[0];
+    const [vault] = await deployOffChainAssetVault();
+
+    await vault.grantRole(await vault.ERC20SNAPSHOTTER(), alice.address);
+
+    const { id } = (await getEventArgs(
+      await vault.snapshot(),
+      "Snapshot",
+      vault
+    )) as SnapshotEvent["args"];
+
+    assert(id.eq(ethers.BigNumber.from(1)), `ID not set`);
+  });
+  it("Sets correct erc20Tier and mintier", async function () {
```

Comment:
ok so you set the tier contracts, where do you test that the tiers are required for transferring and confiscating both erc20 and erc1155?